torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -727,39 +889,38 @@ def _pad_with_scalar_fill(
     shape = image.shape
     num_channels, height, width = shape[-3:]
 
-    if image.numel() > 0:
-        image = image.reshape(-1, num_channels, height, width)
-
-        if padding_mode == "edge":
-            # Similar to the padding order, `torch_pad`'s PIL's padding modes don't have the same names. Thus, we map
-            # the PIL name for the padding mode, which we are also using for our API, to the corresponding `torch_pad`
-            # name.
-            padding_mode = "replicate"
-
-        if padding_mode == "constant":
-            image = torch_pad(image, torch_padding, mode=padding_mode, value=float(fill))
-        elif padding_mode in ("reflect", "replicate"):
-            # `torch_pad` only supports `"reflect"` or `"replicate"` padding for floating point inputs.
-            # TODO: See https://github.com/pytorch/pytorch/issues/40763
-            dtype = image.dtype
-            if not image.is_floating_point():
-                needs_cast = True
-                image = image.to(torch.float32)
-            else:
-                needs_cast = False
-
-            image = torch_pad(image, torch_padding, mode=padding_mode)
-
-            if needs_cast:
-                image = image.to(dtype)
-        else:  # padding_mode == "symmetric"
-            image = _FT._pad_symmetric(image, torch_padding)
+    batch_size = 1
+    for s in shape[:-3]:
+        batch_size *= s
 
-        new_height, new_width = image.shape[-2:]
-    else:
-        left, right, top, bottom = torch_padding
-        new_height = height + top + bottom
-        new_width = width + left + right
```

Comment:
@pmeier This won't work with pad.

The following test will fail if on `crop_image_tensor` we use this version of pad instead of the stable:
```
test/test_prototype_transforms_functional.py:236 (TestKernels.test_against_reference[crop_image_tensor-024])
Traceback (most recent call last):
  File "./vision/test/test_prototype_transforms_functional.py", line 241, in test_against_reference
    actual = info.kernel(input, *other_args, **kwargs)
  File "./vision/torchvision/prototype/transforms/functional/_geometry.py", line 1046, in crop_image_tensor
    return pad_image_tensor(image[..., max(top, 0) : bottom, max(left, 0) : right], padding_ltrb)
  File "./vision/torchvision/prototype/transforms/functional/_geometry.py", line 876, in pad_image_tensor
    return _pad_with_scalar_fill(image, torch_padding, fill=fill, padding_mode=padding_mode)
  File "./vision/torchvision/prototype/transforms/functional/_geometry.py", line 926, in _pad_with_scalar_fill
    return image.reshape(shape[:-3] + (num_channels, new_height, new_width))
RuntimeError: shape '[1, 12, 12]' is invalid for input of size 0
```